Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1022,10 +1022,11 @@ private bool ShouldSandboxedProcessExecuteExternal
                 {
                     Contract.Assert(m_sandboxConfig.AdminRequiredProcessExecutionMode == AdminRequiredProcessExecutionMode.ExternalVM);
 
-                    Tracing.Logger.Log.PipProcessStartExternalVm(m_loggingContext, m_pip.SemiStableHash, m_pip.GetDescription(m_context));
-
+                    // Initialize VM once.
                     await m_vmInitializer.LazyInitVmAsync.Value;
 
+                    Tracing.Logger.Log.PipProcessStartExternalVm(m_loggingContext, m_pip.SemiStableHash, m_pip.GetDescription(m_context));

```

Comment:
Why you need to move the logging here? what is the difference?

Reply:
The logging is a bit confusing before. It shows that the VM execution starts before initialization happens. So, this ordering should remove that confusion.